### PR TITLE
feat: Move use case `CreateCandidate` to new profile

### PR DESCRIPTION
**Pull request by martinalbert**
## Description
- moves use case `CreateCandidate` to new profile `recruitment/create-candidate`
- subsequently removes profile `recruitment/candidates`

## Motivation and Context
Discussion in [this comment](https://github.com/superfaceai/station/pull/305#discussion_r1048169804), main reason was different support from providers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by janhalama — APPROVED** [review]
LGTM

Note: Removing `recruitment/candidates` from station doesn't remove the profile from registry. I miss flag obsolete or deprecated in registry to prevent listing the profile in catalog.

**Comment by martinalbert** [comment]
Should I keep this PR open until we deprecate `recruitment/candidates` profile?

**Comment by janhalama** [comment]
> Should I keep this PR open until we deprecate `recruitment/candidates` profile?

Catalog and registry do not support deprecating of profiles yet. We can update `recruitment/candidates` profile description. It can be part of this PR.

File: grid/recruitment/candidates/profile.supr [hunk 1]
```diff
@@ -1,493 +1,10 @@
 """
-Manage Candidates
+Manage Candidates 
 Fetch, list and create candidates in ATS (Applicant Tracking System).
+
+@deprecated 
+Use the recruitment/create-candidate or other profiles instead.
 """
 name = "recruitment/candidates"
 version = "1.0.0"
```
Comment (janhalama) [inline grid/recruitment/candidates/profile.supr:9]:
```suggestion
version = "1.0.1"
```
Patch version needs to be bumped.

File: grid/recruitment/candidates/profile.supr [hunk 2]
```diff
@@ -1,493 +1,10 @@
...
 
-"""
```
[the remark this answers is not in the document]
Comment (martinalbert) [reply grid/recruitment/candidates/profile.supr:8]: What about maps, should we keep all of them and return error that it is deprecated or just a mock?

[the remark this answers is not in the document]
Comment (janhalama) [reply grid/recruitment/candidates/profile.supr:8]: Maps are written for major.minor version of profile:
```
profile = "recruitment/candidates@1.0"
```

Previously published maps in registry will work well with new version `1.0.1` of `recruitment/candidates`  profile.

[the remark this answers is not in the document]
Comment (janhalama) [reply grid/recruitment/candidates/profile.supr:8]: We can delete the maps.

[the remark this answers is not in the document]
Comment (martinalbert) [reply grid/recruitment/candidates/profile.supr:8]: done 👍🏻

**Review by janhalama — APPROVED** [review]
LGTM

**Pull request merged**